components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java

```diff
@@ -217,7 +219,16 @@ public String renderFragment(HttpRequest request, HttpResponse response) {
         Model model = new MapModel(request.getFormParams());
         RequestLookup requestLookup = createRequestLookup(request, response);
         API api = new API(sessionRegistry, requestLookup);
-        return fragment.render(model, lookup, requestLookup, api);
+
+        JsonObject renderedFragment = new JsonObject();
+        renderedFragment.addProperty("html", fragment.render(model, lookup, requestLookup, api));
+        renderedFragment.addProperty(Placeholder.headJs.name(),
```

Comment:
We are returning a JSON here (`JsonObject`). IMO, creating a new class (with setJs, setCss etc.) just for this method is not a good way